lib/ansible/modules/monitoring/zabbix/zabbix_action.py

```diff
@@ -0,0 +1,1653 @@
+    name = module.params['name']
+    esc_period = module.params['esc_period']
+    event_source = module.params['event_source']
+    state = module.params['state']
+    status = module.params['status']
+    default_message = module.params['default_message']
+    default_subject = module.params['default_subject']
+    recovery_default_message = module.params['recovery_default_message']
+    recovery_default_subject = module.params['recovery_default_subject']
+    acknowledge_default_message = module.params['acknowledge_default_message']
+    acknowledge_default_subject = module.params['acknowledge_default_subject']
+    conditions = module.params['conditions']
+    formula = module.params['formula']
+    operations = module.params['operations']
+    recovery_operations = module.params['recovery_operations']
+    acknowledge_operations = module.params['acknowledge_operations']
+
+    try:
+        zbx = ZabbixAPI(server_url, timeout=timeout, user=http_login_user,
+                        passwd=http_login_password, validate_certs=validate_certs)
+        zbx.login(login_user, login_password)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Zabbix server: %s" % e)
+
+    zapi_wrapper = Zapi(module, zbx)
+
+    action = Action(module, zbx, zapi_wrapper)
+
+    action_exists = zapi_wrapper.check_if_action_exists(name)
+    ops = Operations(module, zbx, zapi_wrapper)
+    recovery_ops = RecoveryOperations(module, zbx, zapi_wrapper)
+    acknowledge_ops = AcknowledgeOperations(module, zbx, zapi_wrapper)
+    fltr = Filter(module, zbx, zapi_wrapper)
+
+    if action_exists:
+        action_id = zapi_wrapper.get_action_by_name(name)['actionid']
+        if state == "absent":
+            result = action.delete_action(action_id)
+            module.exit_json(changed=True, result="Action Deleted: %s, ID: %s" % (name, result))
+        else:
+            difference = action.check_difference(
+                action_id=action_id,
+                name=name,
+                event_source=event_source,
+                esc_period=esc_period,
+                status=status,
+                default_message=default_message,
+                default_subject=default_subject,
+                recovery_default_message=recovery_default_message,
+                recovery_default_subject=recovery_default_subject,
+                acknowledge_default_message=acknowledge_default_message,
+                acknowledge_default_subject=acknowledge_default_subject,
+                operations=ops.construct_the_data(operations),
+                recovery_operations=recovery_ops.construct_the_data(recovery_operations),
+                acknowledge_operations=acknowledge_ops.construct_the_data(acknowledge_operations),
+                conditions=fltr.construct_the_data(formula, conditions)
+            )
+
+            if difference == {}:
+                module.exit_json(changed=False, result="Action is up to date: %s" % (name))
```

Comment:
I would use the key of the common return `msg` instead of `result`, as it is basically a message

https://docs.ansible.com/ansible/2.7/reference_appendices/common_return_values.html#msg

Reply:
Agree. Will replace it today.

Reply:
Done.